internal/pkg/composable/providers/host/host.go

```diff
@@ -49,13 +49,26 @@ func (c *contextProvider) Run(comm corecomp.ContextProviderComm) error {
 		return errors.New(err, "failed to set mapping", errors.TypeUnexpected)
 	}
 
+	const fqdnFeatureFlagCallbackID = "host_provider"
+	fqdnFFChangeCh := make(chan struct{})
+	err = features.AddFQDNOnChangeCallback(
+		onFQDNFeatureFlagChange(fqdnFFChangeCh),
+		fqdnFeatureFlagCallbackID,
+	)
+	if err != nil {
+		return fmt.Errorf("unable to add FQDN onChange callback in host provider: %w", err)
+	}
+
 	// Update context when any host information changes.
 	for {
 		t := time.NewTimer(c.CheckInterval)
 		select {
 		case <-comm.Done():
 			t.Stop()
+			features.RemoveFQDNOnChangeCallback(fqdnFeatureFlagCallbackID)
+			close(fqdnFFChangeCh)
```

Comment:
Please move to a defer on line 61 it would be cleaner in the case that a future change results in it not being unregistered.

```golang
defer func() {
   features.RemoveFQDNOnChangeCallback(fqdnFeatureFlagCallbackID)
   close(fqdnFFChangeCh)
}()

Reply:
Good point, thanks.  Done in c008e54d5.